JEMS-CSE416/JEMS
# Pagination

** IMPORTANT **
DO NOT MERGE THIS REQUEST... if [DuplicateMap rolling out extra features](https://github.com/JEMS-CSE416/JEMS/pull/75) is  NOT MERGED. This pr is dependent on the merge of the pr mentioned above.

**Updates**
- cleaned up code
- Implemented pagination of my maps, discover, and searched maps screen.
- Added a new custom hook called useLoadingData that lets you retrieve data but also lets you know if it's still loading. useful for letting the user know that the data is still loading.


**How to review:**
```
git checkout main
git fetch
git checkout pagination
cd client
npm start
```

1. Go to browser & see something like below: 
<img width="1792" alt="Screenshot 2023-11-25 at 4 37 40 PM" src="https://github.com/JEMS-CSE416/JEMS/assets/76068901/2da62776-80f3-45c5-b14d-42d7940edd71">

2. after a few seconds the browser should display the maps:
<img width="1792" alt="Screenshot 2023-11-25 at 4 38 45 PM" src="https://github.com/JEMS-CSE416/JEMS/assets/76068901/03790d59-9cc8-408d-8431-3eaac2b4fe10">

3. head over to my maps and the same should occur.
<img width="1792" alt="Screenshot 2023-11-25 at 4 39 08 PM" src="https://github.com/JEMS-CSE416/JEMS/assets/76068901/afc6f087-811a-4443-8f62-5c2f3de073c5">

4. head over to discover maps and the same should also happen:
<img width="1792" alt="Screenshot 2023-11-25 at 4 40 23 PM" src="https://github.com/JEMS-CSE416/JEMS/assets/76068901/27065665-e4b4-4e6b-b838-db797a81efa3">

5. finally try to click on different pages via the pagination navigation at the bottom.
